packages/core/src/components/forms/_numeric-input.scss

```diff
@@ -0,0 +1,208 @@
+
+@import "../../common/variables";
+@import "./common";
+
+/*
+Numeric inputs
+
+Numeric inputs provide controls for inputting, incrementing, and decrementing a numeric value.
+
+##### Keyboard interactions
+
+- `↑/↓` - change the value by one step (default: `1`)
+- `Shift + ↑/↓` - change the value by one major step (default: `10`)
+- `Alt + ↑/↓` - change the value by one minor step (default: `0.1`)
+- `Enter` - triggers the `onConfirm` callback if provided; otherwise, sets invalid input to `null`
+
+##### Mouse interactions
+
+- `Click ↑/↓` - change the value by one step (default: `1`)
+- `Shift + Click ↑/↓` - change the value by one major step (default: `10`)
+- `Alt + Click ↑/↓` - change the value by one minor step (default: `0.1`)
+
+##### Example
+
+The following example is a configurable `NumericInput` that responds to the keyboard and mouse interactions listed above.
+
+@react-example NumericInputExample
+
+Weight: 10
+
+Styleguide components.forms.numeric-input
+*/
+
+/*
+JavaScript API
+
+The `NumericInput` component is available in the __@blueprintjs/core__ package.
+Make sure to review the [general usage docs for JS components](#components.usage).
+
+A numeric input can function as either an uncontrolled component or a semi-controlled component. It cannot
+function as a purely controlled component, because it needs to adjust its own copy of the `value` internally
+in response to keyboard and mouse interactions.
+
+##### Uncontrolled mode
+
+To use the numeric input in uncontrolled mode, simply pass an `onUpdate` callback into the component
+to extract the value as the component changes it in response to keyboard and/or mouse interactions. The
+component will invoke the callback after every value change, giving you the new value as a `string`.
```

Comment:
Changed to `number | string` with the return scheme: if `this.isValueNumeric(this.state.value)`, then return `parseFloat(value)`, else return the `value` as a `string`. How does that sound?